src/mono/netcore/System.Private.CoreLib/src/System/Collections/Generic/Comparer.Mono.cs

```diff
@@ -8,24 +8,13 @@ namespace System.Collections.Generic
 {
 	partial class Comparer<T>
 	{
-        static volatile Comparer<T> defaultComparer;
-
-		public static Comparer<T> Default {
-			get {
-                Comparer<T> comparer = defaultComparer;
-                if (comparer == null) {
-                    comparer = CreateComparer();
-                    defaultComparer = comparer;
-                }
-                return comparer;
-			}
-		}
+		public static Comparer<T> Default { get; } = CreateComparer();
```

Comment:
This generates static ctor, considering this type is generic and abstract this could hurt AOT. Could you use CAS instead?

/cc @vargaz for double-checking this construction is still problematic

Reply:
> Could you use CAS instead?

Done